### PR TITLE
Upgrade to curve25519-dalek-ng 4.1 and subtle-ng 2.5

### PR DESCRIPTION
**Pull request by delta1**
- upgrade to curve25519-dalek-ng 4.1 
- upgrade to subtle-ng 2.5 
- bump version to 4.1 to match dalek
- fix doctests in r1cs 
- upgrade doc includes to new stable syntax 

tested on latest stable rust 1.55, and nightly-2021-09-15 for nightly features

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,3 @@
 /target/
 **/*.rs.bk
-Cargo.lock
+# Cargo.lock
```
Comment (CjS77) [inline .gitignore]: Typically libraries don't check in Cargo.lock.

Relevant advice from Rust docs:
"If you’re building a non-end product, such as a rust library that other rust packages will depend on, put Cargo.lock in your .gitignore"

https://doc.rust-lang.org/cargo/guide/cargo-toml-vs-cargo-lock.html

File: docs/r1cs-docs-example.md [hunk 2]
```diff
@@ -13,7 +13,7 @@ Algebraically it can be expressed as a statement that for a free variable \\(z\\
 The prover can commit to blinded scalars \\(x_i\\) and \\(y_i\\) then receive a random challenge \\(z\\),
 and build a proof that the above relation holds.
 
-K-shuffle requires \\( 2*(K-1) \\) multipliers.
+K-shuffle requires \\( 2\*(K-1) \\) multipliers.
```
Comment (CjS77) [inline docs/r1cs-docs-example.md]:
```suggestion
K-shuffle requires \\( 2(K-1) \\) multipliers.
```

**Review by CjS77 — COMMENTED** [review]
Looks good, but you typically don't commit Cargo.lock in libraries

File: .gitignore (hunk 1 above)
Comment (delta1) [reply .gitignore]: ah that's why, cool will update

File: docs/r1cs-docs-example.md (hunk 2 above)
Comment (delta1) [reply docs/r1cs-docs-example.md]: sorry this was an auto fmt

**Review by stringhandler — APPROVED** [review]

**Pull request merged**